Update CHANGELOG.md v4.70.0 (#14910)
hashicorp/terraform-provider-google: CHANGELOG.md

## 4.70.0 (Unreleased)

FEATURES:
* **New Resource:** `google_compute_network_endpoints` ([#14869](https://github.com/hashicorp/terraform-provider-google/pull/14869))
* **New Resource:** `vertex_ai_index_endpoint` ([#14842](https://github.com/hashicorp/terraform-provider-google/pull/14842))

IMPROVEMENTS:

> * bigtable: added timeout support for bigtable admin operations ([#14861](https://github.com/hashicorp/terraform-provider-google/pull/14861))

Author: Updated.

* cloudfunctions2: added `url` output field to `google_cloudfunctions2_function` ([#14851](https://github.com/hashicorp/terraform-provider-google/pull/14851))
* compute: added field `network_attachment` to `google_compute_instance_template` ([#14874](https://github.com/hashicorp/terraform-provider-google/pull/14874))

> * compute: surface error details for Compute Operation with quota exceeded errors. ([#14879](https://github.com/hashicorp/terraform-provider-google/pull/14879))

Reviewer: This would be good to rewrite to be more focused on the customer experience - that is, not referring to API objects like operations.

Author: Looks like this changed `computed_operation`, I'm not sure exactly which operations does it refer here.

Reviewer: Resolving this thread since it's outdated & not showing up on the changed view any more; I'll open a new thread with a suggestion.

> * compute: added error details for Compute Operation with quota exceeded errors. ([#14879](https://github.com/hashicorp/terraform-provider-google/pull/14879))

Author: Sounds good, thanks for the suggestion.

> * compute: added pathTemplateMatch and pathTemplateRewrite to routeRules in UrlMap. ([#14873](https://github.com/hashicorp/terraform-provider-google/pull/14873))

Author: Updated

> * compute: added in ability to update Hyperdisk PD IOPS without recreation ([#14844](https://github.com/hashicorp/terraform-provider-google/pull/14844))

Reviewer: This should include the specific names of impacted resources

* container: added `sole_tenant_config` to `node_config` in `google_container_node_pool` and `google_container_cluster` ([#14897](https://github.com/hashicorp/terraform-provider-google/pull/14897))
* dataform: added field `workspace_compilation_overrides` to resource `google_dataform_repository` (beta) ([#14839](https://github.com/hashicorp/terraform-provider-google/pull/14839))
* dlp: added `crypto_hash_config` to `google_data_loss_prevention_deidentify_template` ([#14870](https://github.com/hashicorp/terraform-provider-google/pull/14870))
* dlp: added `trigger_id` field to `google_data_loss_prevention_job_trigger` ([#14892](https://github.com/hashicorp/terraform-provider-google/pull/14892))
* dlp: added missing file types `POWERPOINT` and `EXCEL` in `inspect_job.storage_config.cloud_storage_options.file_types` enum to `google_data_loss_prevention_job_trigger` resource ([#14856](https://github.com/hashicorp/terraform-provider-google/pull/14856))

> * dlp: added multiple `sensitivity_score` field to `google_data_loss_prevention_deidentify_template` resource ([#14880](https://github.com/hashicorp/terraform-provider-google/pull/14880))

Author: The name of the field is `sensitivity_score`, added in multiple places.. I would leave it like this for now..

> * dlp: added multiple `sensitivity_score` field to `google_data_loss_prevention_deidentify_template` resource ([#14880](https://github.com/hashicorp/terraform-provider-google/pull/14880))

Reviewer: SGTM

* dlp: added multiple `sensitivity_score` field to `google_data_loss_prevention_inspect_template` resource ([#14871](https://github.com/hashicorp/terraform-provider-google/pull/14871))
* dlp: added multiple `sensitivity_score` field to `google_data_loss_prevention_job_trigger` resource ([#14881](https://github.com/hashicorp/terraform-provider-google/pull/14881))
* dlp: changed `inspect_template_name` field from required to optional in `google_data_loss_prevention_job_trigger` resource ([#14845](https://github.com/hashicorp/terraform-provider-google/pull/14845))

> * pubsub: allowed `definition` of `google_pubsub_schema` to change without deleting and recreating the resource by using schema revisions (https://cloud.google.com/pubsub/docs/schemas#commit-schema-revision) ([#14857](https://github.com/hashicorp/terraform-provider-google/pull/14857))

Reviewer: This should be rewritten to avoid discussing implementation details

Author: Updated

> * sql: added `POSTGRES_15` to available version docs ([#14891](https://github.com/hashicorp/terraform-provider-google/pull/14891))

Reviewer: This should be rewritten to include the resource / field names impacted

Author: Updated

> * vpcaccess: Added `connected_projects` attribute to resource `google_vpc_access_connector`. ([#14835](https://github.com/hashicorp/terraform-provider-google/pull/14835))

Reviewer: ```suggestion
* vpcaccess: added `connected_projects` field to resource `google_vpc_access_connector`. ([#14835](https://github.com/hashicorp/terraform-provider-google/pull/14835))
```

Author: Thanks for the suggestion, updated

BUG FIXES:

> * filestore: fixed an issue causing 429 errors to be retried ([#14850](https://github.com/hashicorp/terraform-provider-google/pull/14850))

Reviewer: this should name the impacted resource

Reviewer: I agree, 7-8 resources is too many to include in one note. in that case how about something like: 

```suggestion
* provider: fixed an issue on multiple resources where non-retryable quota errors were considered retryable ([#14850](https://github.com/hashicorp/terraform-provider-google/pull/14850))
```

Or you could have one note per resource - that might be better:

```suggestion
* filestore: fixed an issue on `google_filestore_RESOURCE` where non-retryable quota errors were considered retryable ([#14850](https://github.com/hashicorp/terraform-provider-google/pull/14850))
```

And then repeat that per resource

> * vertexai: made google_vertex_ai_featurestore_entitytype_feature always use regional corresponding to parent's region ([#14843](https://github.com/hashicorp/terraform-provider-google/pull/14843))

Reviewer: ```suggestion
* vertexai: made `google_vertex_ai_featurestore_entitytype_feature` always use region corresponding to parent's region ([#14843](https://github.com/hashicorp/terraform-provider-google/pull/14843))
```

Author: Thanks for the suggestion, updated
